### PR TITLE
Add links redirect inside channel description

### PR DESCRIPTION
**Pull request by MathiusD**
The purpose of this PR is to be able to follow the links within the description of channel (Like this one for example : https://yewtu.be/channel/UCXF0YCBWewAj3RytJUAivGA)

### CODE REVIEW

**Comment by SamantazFox** [comment]
Mmmmh, we already have a function that's supposed to transforms URLs in comments and descriptions.

Youtube should return the descriptions in an array similar to this:
```json
"runs": [
  {
    "text": "Welcome to LabPadre, an intelligent chat community! [SNIP]\n"
  },
  {
    "text": "https://www.islagrand.com",
    "navigationEndpoint": {
      // SNIP
      "urlEndpoint": {
        "url": "https://www.youtube.com/[SNIP]",
        "target": "TARGET_NEW_WINDOW",
        "nofollow": true
      }
    }
  },
  {
    "text": "\nVisit Gators Dockside Online: "
  },
  // SNIP
]
```

**Comment by MathiusD** [comment]
> Mmmmh, we already have a function that's supposed to transforms URLs in comments and descriptions.
> 
> Youtube should return the descriptions in an array similar to this:
> 
> ```json
> "runs": [
>   {
>     "text": "Welcome to LabPadre, an intelligent chat community! [SNIP]\n"
>   },
>   {
>     "text": "https://www.islagrand.com",
>     "navigationEndpoint": {
>       // SNIP
>       "urlEndpoint": {
>         "url": "https://www.youtube.com/[SNIP]",
>         "target": "TARGET_NEW_WINDOW",
>         "nofollow": true
>       }
>     }
>   },
>   {
>     "text": "\nVisit Gators Dockside Online: "
>   },
>   // SNIP
> ]
> ```

I presume you are referring to the `content_to_comment_html` in `comments.cr` ? (I confess I didn't use it because I didn't know if it was adapted here)

**Comment by SamantazFox** [comment]
> I presume you are referring to the `content_to_comment_html` in `comments.cr` ? (I confess I didn't use it because I didn't know if it was adapted here)

Yep, that's the one! It is used for video and playlist descriptions, as well as comments. It should also have been used for the channel description too.

The problem seem to come from here:
https://github.com/iv-org/invidious/blob/47690fffc116da419c2d212e68ef6df5b1cc8927/src/invidious/channels/about.cr#L54-L55
and here:
https://github.com/iv-org/invidious/blob/47690fffc116da419c2d212e68ef6df5b1cc8927/src/invidious/channels/about.cr#L74-L75

File: src/invidious/comments.cr [hunk 1]
```diff
@@ -554,6 +554,32 @@ def fill_links(html, scheme, host)
   return html.to_xml(options: XML::SaveOptions::NO_DECL)
 end
 
+def text_to_parsed_content(text : String) : JSON::Any
```
Comment (SamantazFox) [inline src/invidious/comments.cr:557]: Can you please lighten the code a bit (space out the different chunks) and add comments, please?

**Comment by SamantazFox** [comment]
Ooops, sorry for the delay. Can you rebase, please?

**Comment by MathiusD** [comment]
If I haven't done anything wrong, it's done @SamantazFox !

**Review by SamantazFox — APPROVED** [review]
Code looks good to me :)

**Comment by SamantazFox** [comment]
Deployed on test.invidious.io

**Comment by SamantazFox** [comment]
thanks for your contribution to invidious :)

**Pull request merged**